testnet/stacks-node/src/neon_node.rs

```diff
@@ -155,6 +158,31 @@ struct MiningTenureInformation {
     coinbase_nonce: u64,
 }
 
+/// Generate the block-builder settings from the node config
+fn make_block_builder_settings(config: &Config, attempt: u64) -> BlockBuilderSettings {
+    BlockBuilderSettings {
+        execution_cost: config.block_limit.clone(),
+        max_miner_time_ms: if attempt <= 1 {
+            // first attempt to mine a block -- do so right away
+            config.miner.first_attempt_time_ms
+        } else {
+            // second or later attempt to mine a block -- give it some time
+            config.miner.subsequent_attempt_time_ms
+        },
+        mempool_settings: MemPoolWalkSettings {
+            min_tx_fee: config.miner.min_tx_fee,
+            min_cumulative_fee: config.miner.min_cumulative_tx_fee,
+            max_walk_time_ms: if attempt <= 1 {
+                // first attempt to mine a block -- do so right away
+                config.miner.first_attempt_time_ms
+            } else {
+                // second or later attempt to mine a block -- give it some time
+                config.miner.subsequent_attempt_time_ms
+            },
+        },
+    }
+}
+
```

Comment:
I think this makes more sense as a method of `BlockBuilderSettings`, e.g., `BlockBuilderSettings::from_config` or the like, and separated from the `neon_node` module.

Reply:
You can't `impl` a method for a struct in a different crate.

Reply:
Right -- I think this code shouldn't be in `neon_node` module (or stacks-node crate) -- it should live with the BlockBuilderSettings. But I guess that'd introduce a circular dependency on `Config`, so maybe just move this to `config.rs` and make it a member of `Config` like `config.make_block_builder_settings(attempt)`.